services/api/src/resources/task/helpers.ts

```diff
@@ -168,54 +140,174 @@ export const Helpers = (sqlClientPool: Pool) => ({
+          projectKeyInjection,
+          adminOnlyView,
+          remoteId,
+          type: 'advanced',
+          advanced_image: image,
+          advanced_payload: JSON.stringify(payload),
+        }),
+      );
 
-    try {
-      await createMiscTask(
-        {
-          key: ADVANCED_TASK_EVENT_TYPE,
-          data: jobSpec
+      const { insertId } = queryresp;
+      rows = await query(sqlClientPool, Sql.selectTask(insertId));
+      const taskData = R.prop(0, rows);
+      const ADVANCED_TASK_EVENT_TYPE = "task:advanced"
+
+      taskData.id = taskData.id.toString();
+
+      let jobSpec = {
+        task: taskData,
+        project: projectData,
+        environment: environmentData,
+        advancedTask: {
+          RunnerImage: image,
+          JSONPayload: new Buffer(JSON.stringify(payload).replace(/\\n/g, "\n")).toString('base64'),
+          deployerToken: deployTokenInjection, //an admintask will have a deployer token and ssh key injected into it
+          sshKey: projectKeyInjection,
         }
-      )
-    } catch (error) {
-      sendToLagoonLogs(
-        'error',
-        projectData.name,
-        '',
-        'api:addTask',
-         { taskId: taskData.id },
-        `*[${projectData.name}]* Task not initiated, reason: ${error}`,
+      }
+
+      try {
+        await createMiscTask(
+          {
+            key: ADVANCED_TASK_EVENT_TYPE,
+            data: jobSpec
+          }
+        )
+      } catch (error) {
+        sendToLagoonLogs(
+          'error',
+          projectData.name,
+          '',
+          'api:addTask',
+          { taskId: taskData.id },
+          `*[${projectData.name}]* Task not initiated, reason: ${error}`,
+        );
+      }
+
+      return taskData;
+    },
+    getTaskByTaskInput: async taskInput => {
+      const notEmpty = R.complement(R.anyPass([R.isNil, R.isEmpty]));
+      const hasId = R.both(R.has('id'), R.propSatisfies(notEmpty, 'id'));
+      const hasName = R.both(R.has('taskName'), R.propSatisfies(notEmpty, 'taskName'));
+      const hasEnvironment = R.both(
+        R.has('environment'),
+        R.propSatisfies(notEmpty, 'environment')
       );
-    }
+      // @ts-ignore
+      const hasNameAndEnvironment = R.both(hasName, hasEnvironment);
+
+      const taskFromId = asyncPipe(
+          R.prop('id'),
+          getTaskById,
+          task => {
+            if (!task) {
+              throw new Error('Unauthorized');
+            }
 
-    return taskData;
-  },
-});
+            return task;
+          }
+        );
+
+        const taskFromNameEnv = async input => {
+          const environments = await environmentHelpers(
+            sqlClientPool
+          ).getEnvironmentsByEnvironmentInput(R.prop('environment', input));
+          const activeEnvironments = R.filter(
+            R.propEq('deleted', '0000-00-00 00:00:00'),
+            environments
+          );
+
+          if (activeEnvironments.length < 1 || activeEnvironments.length > 1) {
```

Comment:
`activeEnvironments.length != 1` maybe?

Also, I'm a little puzzled by the error - is "Unauthorized" appropriate? (it might be, of course! Just wanted to check)